### PR TITLE
Prob compare

### LINKED ISSUE #50: Deprecate `_load_xarray()` and `_load_raster_as_xarray()`
## Current behavior
- `_load_raster_as_xarray()` is just a wrapper for `rxr.open_rasterio`
- `_load_xarray` does have added benefit of loading from s3. Could use more descriptive name and used with an alias?

### PR DESCRIPTION
**Pull request by fernando-aristizabal**
Probabilistic metrics
1. added xskillscore as dependency for prob metrics
2. updated `_assert_pairing_dict_equal()` to use `deep diff.DeepDiff()` function
    - addition of xskillscore slightly changed some values for continuous metrics.
3. updated tests to use `Prob_metrics_df` schema for `expected_df`s
4. stopped coercion for `Prob_metrics_df` schema
5. ran `black` and `flake8`
6. updated crps_ensemble test to use correct size variable
7. added probabilistic metrics via `compute_probabilistic_metrics.py`
8. added `probabilistic_compare()` as gval accessor method
9. added `_compare_metrics_df_with_xarray()` to compare df's with xarray objects
10. added probabilistic compare accessor method
12. updated docstrings to represent correct module name or function
13. deprecated `load_raster_as_xarray()`. closes #50
14. added `_create_circle_mask()`, `_create_xarray()`, and `_create_xarray_pairs()`
15. added `Prob_metrics_df` dataframe schema

### CODE REVIEW

File: src/gval/utils/loading_datasets.py [hunk 1]
```diff
@@ -320,3 +227,230 @@ def _convert_to_dataset(xr_object=Union[xr.DataArray, xr.Dataset]) -> xr.Dataset
...
+        Xarray with the circles per band.
+    """
+
+    # Handle sizes
+    if isinstance(sizes, Number):
+        sizes = int(sizes)
+        sizes = (sizes, sizes)
+
+    # handle shapes
+    if shapes == "circle":
```
Comment (GregoryPetrochenkov-NOAA) [inline src/gval/utils/loading_datasets.py:319]: I would remove 318-325  and delegate check for valid shape to the loop starting 330 unless this is for memory purposes or some future masks may generalize to multiple shapes.

File: tests/cases_compute_probabilistic_metrics.py [hunk 2]
```diff
@@ -0,0 +1,821 @@
...
+        (100, 300, (20, 20), 7),
+        (150, 350, (30, 30), 10),
+        (200, 400, (40, 40), 8),
+    ]
+
+    # Band parameters for benchmark
+    # background value, circle value, circle center, and circle radius
+    band_params_benchmark = [
+        (145, 360, (25, 25), 15),
+        # (160, 360, (2, 2), 3),
```
Comment (GregoryPetrochenkov-NOAA) [inline tests/cases_compute_probabilistic_metrics.py:105]: Unless there is a purpose remove commented lines.

**Review by GregoryPetrochenkov-NOAA — CHANGES_REQUESTED** [review]
Just some minor changes, some other things I am going to open up as issues.

File: tests/cases_load_xarray.py [hunk 3]
```diff
@@ -24,3 +26,305 @@
...
+        (0, 1, (7, 7), 3),  # band 1
+        (0, 2, (7, 7), 3),  # band 2
+    ]
+
+    band_params_benchmark = [
+        (0, 2, (7, 7), 3),  # band 1
+        (0, 1, (7, 7), 3),  # band 2
+    ]
+
+    data_band_1 = np.array(
```
[the remark this answers is not in the document]
Comment (fernando-aristizabal) [reply tests/cases_load_xarray.py:191]: All repeated variables in this module were moved to global status. Changes in those values at the function level remain in place.

File: src/gval/utils/loading_datasets.py (hunk 1 above)
Comment (fernando-aristizabal) [reply src/gval/utils/loading_datasets.py:319]: Yes, the purpose of this is to support other array creation functions eventually.

File: tests/cases_compute_probabilistic_metrics.py (hunk 2 above)
Comment (fernando-aristizabal) [reply tests/cases_compute_probabilistic_metrics.py:105]: Removed.

**Review by GregoryPetrochenkov-NOAA — APPROVED** [review]

**Pull request merged**